### PR TITLE
Wrap ArgumentException with CryptographicException for incorrect key in EnvelopedCms

### LINKED ISSUE #50222: Test failure System.Security.Cryptography.Pkcs.EnvelopedCmsTests.Tests.DecryptTestsUsingCertWithPrivateKey.DecryptUsingCertificateWithSameSubjectKeyIdentifierButDifferentKeyPair
Run: [runtime-coreclr libraries-jitstress 20210324.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1053900&view=ms.vss-test-web.build-test-results-tab&runId=32540382&resultId=152378&paneView=debug)

Failed test:
```
net6.0-Linux-Release-x64-CoreCLR_checked-zapdisable-Ubuntu.1804.Amd64.Open
 -System.Security.Cryptography.Pkcs.EnvelopedCmsTests.Tests.DecryptTestsUsingCertWithPrivateKey.DecryptUsingCertificateWithSameSubjectKeyIdentifierButDifferentKeyPair
```

**Error message:**
```
Assert.Throws() Failure
Expected: typeof(System.Security.Cryptography.CryptographicException)
Actual:   typeof(System.ArgumentException): Specified key is not a valid size for this algorithm. (Parameter 'rgbKey')
---- System.ArgumentException : Specified key is not a valid size for this algorithm. (Parameter 'rgbKey')


Stack trace
   at Internal.Cryptography.AesImplementation.CreateTransform(Byte[] rgbKey, Byte[] rgbIV, Boolean encrypting) in /_/src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.cs:line 55
   at Internal.Cryptography.Pal.AnyOS.ManagedPkcsPal.ManagedDecryptorPal.DecryptContent(ReadOnlyMemory`1 encryptedContent, Byte[] cek, AlgorithmIdentifierAsn contentEncryptionAlgorithm, Exception& exception) in /_/src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs:line 173
   at Internal.Cryptography.Pal.AnyOS.ManagedPkcsPal.ManagedDecryptorPal.TryDecryptCore(Byte[] cek, String contentType, Nullable`1 content, AlgorithmIdentifierAsn contentEncryptionAlgorithm, Exception& exception) in /_/src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs:line 108
   at Internal.Cryptography.Pal.AnyOS.ManagedPkcsPal.ManagedDecryptorPal.TryDecrypt(RecipientInfo recipientInfo, X509Certificate2 cert, AsymmetricAlgorithm privateKey, X509Certificate2Collection originatorCerts, X509Certificate2Collection extraStore, Exception& exception) in /_/src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs:line 66
   at System.Security.Cryptography.Pkcs.EnvelopedCms.DecryptContent(RecipientInfoCollection recipientInfos, X509Certificate2Collection extraStore) in /_/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/EnvelopedCms.cs:line 249
----- Inner Stack Trace -----
   at Internal.Cryptography.AesImplementation.CreateTransform(Byte[] rgbKey, Byte[] rgbIV, Boolean encrypting) in /_/src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.cs:line 55
   at Internal.Cryptography.Pal.AnyOS.ManagedPkcsPal.ManagedDecryptorPal.DecryptContent(ReadOnlyMemory`1 encryptedContent, Byte[] cek, AlgorithmIdentifierAsn contentEncryptionAlgorithm, Exception& exception) in /_/src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs:line 173
   at Internal.Cryptography.Pal.AnyOS.ManagedPkcsPal.ManagedDecryptorPal.TryDecryptCore(Byte[] cek, String contentType, Nullable`1 content, AlgorithmIdentifierAsn contentEncryptionAlgorithm, Exception& exception) in /_/src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs:line 108
   at Internal.Cryptography.Pal.AnyOS.ManagedPkcsPal.ManagedDecryptorPal.TryDecrypt(RecipientInfo recipientInfo, X509Certificate2 cert, AsymmetricAlgorithm privateKey, X509Certificate2Collection originatorCerts, X509Certificate2Collection extraStore, Exception& exception) in /_/src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs:line 66
   at System.Security.Cryptography.Pkcs.EnvelopedCms.DecryptContent(RecipientInfoCollection recipientInfos, X509Certificate2Collection extraStore) in /_/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/EnvelopedCms.cs:line 249
```

### PR DESCRIPTION
**Pull request by vcsjones**
If an incorrect asymmetric key is used to decrypt a symmetric key in enveloped data, and the decryption is successful but produces an invalid symmetric key, wrap that as a CryptographicException.

Closes #50222.

### CODE REVIEW

**Comment by vcsjones** [comment]
:crossed_fingers: Windows does the same thing.

File: src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTestsUsingCertWithPrivateKey.cs [hunk 1]
```diff
@@ -67,6 +67,31 @@ public static void DecryptMultipleRecipients()
             Assert.Equal<byte>(content, contentInfo.Content);
         }
 
+        [Fact]
+        public static void DecryptSuccesfullyWithWrongKeyProducesInvalidSymmetricKey()
+        {
+            using (X509Certificate2 wrongRecipient = Certificates.RSAKeyTransfer5_ExplicitSkiOfRSAKeyTransfer4.TryGetCertificateWithPrivateKey())
+            {
+                string encryptedContent =
+"3082018806092A864886F70D010703A082017930820175020102318201303082" +
```
Comment (vcsjones) [inline src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTestsUsingCertWithPrivateKey.cs:76]: I produced this by running the originally failing test in a loop until it reproduced the behavior and dumped the enveloped data in the fail case to make a stable test case for this.

File: src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTestsUsingCertWithPrivateKey.cs [hunk 2]
```diff
@@ -67,6 +67,31 @@ public static void DecryptMultipleRecipients()
             Assert.Equal<byte>(content, contentInfo.Content);
         }
 
+        [Fact]
+        public static void DecryptSuccesfullyWithWrongKeyProducesInvalidSymmetricKey()
+        {
+            using (X509Certificate2 wrongRecipient = Certificates.RSAKeyTransfer5_ExplicitSkiOfRSAKeyTransfer4.TryGetCertificateWithPrivateKey())
+            {
+                string encryptedContent =
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTestsUsingCertWithPrivateKey.cs:75]: Want to leave a comment as to how you came up with this payload so someone can replace it/enhance it in the future (trial and error, manually encrypting a facetious payload and patching it in, etc)?

Comment (bartonjs) [reply src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTestsUsingCertWithPrivateKey.cs:75]: Yeah, like https://github.com/dotnet/runtime/pull/50262#discussion_r601872422. But in the code :smile:

**Review by bartonjs — APPROVED** [review]

**Comment by bartonjs** [comment]
Big comment, indeed.  Thanks :smile:.

**Comment by vcsjones** [comment]
> :crossed_fingers: Windows does the same thing.

I guess "building" would be a good place to start :sweat_smile:

**Pull request merged**